subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/DeploymentContinuousBuildIntegrationTest.groovy

```diff
@@ -0,0 +1,145 @@
+                    keyFile.text = key
+                }
+
+                public void start(Deployment deployment) {
+                    running = true
+                }
+
+                public boolean isRunning() {
+                    return running
+                }
+
+                public void stop() {
+                    running = false
+                    keyFile.delete()
+                }
+            }
+
+            class RunTestDeployment extends DefaultTask {
+                @InputFile
+                File triggerFile
+
+                @OutputFile
+                File keyFile
+
+                @Inject
+                protected DeploymentRegistry getDeploymentRegistry() {
+                    throw new UnsupportedOperationException()
+                }
+
+                @TaskAction
+                void doDeployment() {
+                    // we set a different key for every build so we can detect if we
+                    // somehow get a different deployment handle between builds
+                    def key = System.currentTimeMillis()
+
+                    TestDeploymentHandle handle = getDeploymentRegistry().get('test', TestDeploymentHandle.class)
+                    if (handle == null) {
+                        // This should only happen once (1st build), so if we get a different value in keyFile between
+                        // builds then we know we can detect if we didn't get the same handle
+                        handle = getDeploymentRegistry().start('test', DeploymentRegistry.ChangeBehavior.NONE, TestDeploymentHandle.class, key, keyFile)
+                    }
+
+                    println "\\nCurrent Key: \$key, Deployed Key: \$handle.keyFile.text"
+                    assert handle.isRunning()
+                }
+            }
+        """
+        buildTimeout = 30
+    }
+
+    def "deployment promoted to continuous build reports accurate build time" () {
+        triggerFile.text = "0"
+
+        when:
+        withoutContinuousBuild()
+        succeeds("runDeployment")
+
+        then:
+        def key = file('keyFile').text
+        assertDeploymentIsRunning(key)
+        buildTimes.size() == 2
+        buildTimes[0] >= buildTimes[1]
+    }
+
+    def "deployment in continuous build reports accurate build time" () {
+        triggerFile.text = "0"
+
+        when:
+        succeeds("runDeployment")
+
+        then:
+        def key = file('keyFile').text
+        assertDeploymentIsRunning(key)
+
+        when:
+        def lastBuildTime = one(buildTimes)
+        waitBeforeModification triggerFile
+        triggerFile << "\n#a change"
+        succeeds()
+
+        then:
+        assertDeploymentIsRunning(key)
+        lastBuildTime >= one(buildTimes)
+    }
+
+    List<Integer> getBuildTimes() {
+        return (output =~ /BUILD SUCCESSFUL in (\d+)s/).collect { it[1].toString().toInteger() }
+    }
+
+    private static <T> T one(Iterable<T> iterable) {
```

Comment:
I think this is the same as `CollectionUtils.single`

Reply:
Done.